inc/util/stream_utils.hpp

```diff
@@ -40,6 +44,43 @@ namespace ebi
 
         return stream;
     }
+
+    /**
+     * This is a generic method that prints a collection if it has the operations ".size()", ".begin()", ".end()" 
+     * defined and they provide iterators, and the inner type has an overloaded "operator<<"
+     */
+    template <typename T>
+    std::ostream &print_container(std::ostream &os, const T &container) {
+        size_t size = container.size();
+        os << "[";
+        if (size > 0) {
+            auto it = container.begin();
+            os << *it;
+            it++;
+            for (; it != container.end(); ++it) {
+                os << ", " << *it;
```

Comment:
gcc tends to be more "relaxed" with spec validations, thus the reason why Clang was included in Travis. [Their APT repo](http://llvm.org/apt/) seems to be up again!